packages/shared_preferences/shared_preferences/lib/src/devtools_extension.dart

```diff
@@ -0,0 +1,136 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:developer' as developer;
+
+import 'package:flutter/foundation.dart';
+
+import '../shared_preferences.dart';
+
+const String _eventPrefix = 'shared_preferences:';
+
+/// A typedef for the post event function
+@visibleForTesting
+typedef PostEvent = void Function(
+  String eventKind,
+  Map<String, Object?> eventData,
+);
+
+/// A helper class that provides data to the devtool extension.
+///
+/// It is only visible for testing and eval.
+@visibleForTesting
+class DevtoolsExtension {
+  /// The default constructor for [DevtoolsExtension].
+  ///
+  /// Accepts an optional [PostEvent] that should only be overwritten when testing.
+  DevtoolsExtension([this._postEvent = developer.postEvent]);
```

Comment:
can this be a const constructor?

Reply:
Good question, just followed @sigmundch's suggestion. I'll try the const constructor to see if the web runner keeps the library.

Reply:
It can be const, just changed it.